### DISCUSSION TITLE
Can the bundle size be reduced a bit by using features?

### DISCUSSION THREAD

**Discussion by frederikhors** [Byron/google-apis-rs#522] [Q&A]
I was wondering if we can reduce the bundle size a bit if I need to use for example only Google Drive api.

Can we introduce some features to choose from in Cargo.toml?

**Comment by Byron** [comment]
I assume 'bundle size' means the size of the crate as it's uploaded on crates.io.
Since there is the mention of features, I assume the desired outcome is to reduce compile time.

And there I don't think anything can reasonably be done unless one wants to put every group of functions behind feature toggles.
Before any of that could happen, I recommend collecting some numbers with a hand-edited version of the API, showing the performance gains (probably in terms of compilation time, incremental and complete rebuild) before and after.

**Reply by frederikhors** [reply]
> I assume 'bundle size' means the size of the crate as it's uploaded on crates.io.

No, sorry for not being clearer.

I'm referring to the final bundle of my application.

After adding the bare minimum to call the Drive API `files().list()` my project went from 21 MB to 25 MB using `cargo build --release`.

And of course incremental compilation time (the one during development) would also benefit a lot.

**Reply by Byron** [reply]
Great, then I think what I said still applies. For instance, once could use a local vendored version of the crate, remove everything that's not used, and see what that does to the compile times and binary size. That will show what the best-possible gain would be, and from there one could figure out if it's worth having cargo features to divide the crate.
